GitUI/UserControls/RevisionGrid/Graph/RevisionGraph.cs

```diff
@@ -198,53 +199,26 @@ public void HighlightBranch(ObjectId id)
         /// Add a single revision from the git log to the graph, including segments to parents.
         /// </summary>
         /// <param name="revision">The revision to add.</param>
-        /// <param name="insertScore">Insert the (artificial) revision before the node with this score.</param>
-        /// <param name="insertRange">Number of scores "reserved" in the list when inserting.</param>
-        public void Add(GitRevision revision, int? insertScore = null, int insertRange = 0)
+        /// <param name="insertScore">Insert the revision at this score.</param>
+        public void Add(GitRevision revision, int? insertScore = null)
         {
-            // The commits are sorted by the score (not contiuous numbering there may be gaps)
+            // The commits are sorted by the score (not continuous numbering there may be gaps)
             // This commit will be ordered after existing, _maxScore is a preliminary score
-            _maxScore++;
+            // if insertScore is set use that (the scores must be reserved)
+            int nextScore = insertScore ?? ++_maxScore;
 
-            bool updateParents = true;
-            if (!_nodeByObjectId.TryGetValue(revision.ObjectId, out RevisionGraphRevision revisionGraphRevision))
+            if (_nodeByObjectId.TryGetValue(revision.ObjectId, out RevisionGraphRevision revisionGraphRevision))
             {
-                // This revision is added from the log, but not seen before. This is probably a root node (new branch)
-                // OR the revisions are not in topo order. If this the case, we deal with it later.
-                int score = _maxScore;
-
-                if (insertScore is not null && _nodeByObjectId is not null)
-                {
-                    // This revision is to be inserted before a certain node
-                    foreach (var (_, graphRevision) in _nodeByObjectId)
-                    {
-                        if (graphRevision.Score < insertScore)
-                        {
-                            // Lower existing scores to reserve the inserted range
-                            graphRevision.OffsetScore(-insertRange);
-                        }
-                    }
-
-                    score = insertScore.Value - insertRange;
-                }
-
-                revisionGraphRevision = new RevisionGraphRevision(revision.ObjectId, score);
-                _nodeByObjectId.TryAdd(revision.ObjectId, revisionGraphRevision);
+                // This revision was added earlier, but is now found in the log.
+                // Increase the score to the current maxScore to keep the order intact.
+                revisionGraphRevision.EnsureScoreIsAbove(nextScore);
```

Comment:
ditto: The old code has modified `_maxScore`.

Reply:
see above